### PR TITLE
Using HTTPS for search templates where possible for #8482

### PR DESCRIPTION
**Pull request by Debanshu777**
Creating a pr which changes the http call to https

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/search/src/main/assets/searchplugins/daum-kr.xml [hunk 1]
```diff
@@ -6,13 +6,13 @@
 <ShortName>다음지도</ShortName>
 <InputEncoding>utf-8</InputEncoding>
 <Image width="16" height="16">data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAMAAADVRocKAAAARVBMVEUAAAD5+fn5+fn5+fn5+fn5+fn5+fn5+fn5+fn5+fn5+flNkP//1Smn3wAoffz9dG7kWlfjngNhtAAke/0gdv9isiRZrCSQTyACAAAACnRSTlMAwwbS+W9sG8/zKHP4TgAAALhJREFUaN7t2ksOwjAMRdGXfzEQoAH2v1QKE6ARIi0yQvTdmSc5Y0cGYKPxopA30WIodaJWlwDrRDFnEUW1CCOqGXhRzUOUI0CAAAECBL4D5MaO+dyPO1TtqyYA/bKBcu3+Wrn1NM0HmjvNB0peN7St2lTtHiJAgAABAgQIECBAgAABApOB1j5cY9/264s4gdfAH/x4jWYCBAiIECCwSMCLaiv9s4cgqgX90xMkRcElDNlgRCUTLHAB68OktAP7gdIAAAAASUVORK5CYII=</Image>
-<Url type="application/x-suggestions+json" method="GET" template="http://sug.search.daum.net/search_nsuggest">
+<Url type="application/x-suggestions+json" method="GET" template="https://sug.search.daum.net/search_nsuggest">
```
Comment (pocmo) [inline components/browser/search/src/main/assets/searchplugins/daum-kr.xml:9]: This one I can't resolve, but that's true for the `http` version too.

File: components/browser/search/src/main/assets/searchplugins/elebila.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <ShortName>Elebila</ShortName>
   <InputEncoding>UTF-8</InputEncoding>
   <Image height="16" width="16">data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAMAAADVRocKAAAASFBMVEX6+voAAAD6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr6+vr1VbutNXuGkExHTSiMQmuelI4Zvo6SAAAAEXRSTlPhAPAV/NvEvLIDZldAJa6Qin6NnvEAAAGOSURBVGje7drpasMwEARgKc7Z9BpP3b7/m5YGzCYdxNCQLTk0v3f4IDaSTFRq3T2tkZLyNtRa6rBGWhbPtezWSMzipeyRmm3ZIjXLskJuSsFJOtCBDgAdOBPgHCAF4DiHKQDHCDvgBCLtLcoCAunAvQA8igFsR4HDyxgh6QHttAGOErIJ+E4AMathA3CdAGRWpxXwnQBkVqcV8J02wDke0E4boD4hMqYV8B0F9Cc2gO0ogNN8GsB0FOAZAB8LGHma0QCmI4DEA5pHBAgFfMcAuoEo4DsKUNJ4KX0nAKlPH3MmwAPauX8AU8x6QDsewBSzHtCOBzD9BB7QjgU0HmjEb5ke8IJu+l+/Dp0NwHQUiOmjmJOd6QjQCD2gnX8/Ouq0B3zHH9/NM/Ad/QBprOwCuI4COs6YFMB1FIjovtEGtHMFX5kduH7gINw2EGtzEhDhhQEwGwCzATAR0LX5Ag/ZLpw39P9BB47SgQ78PemXBvKvPeyRmtdSN8hIXD2pQ4YQl2cO1382S2Rk9T7U+g3L7mclyX4h2wAAAABJRU5ErkJggg==</Image>
-  <Url type="text/html" template="http://elebila.elhuyar.eus/search/">
+  <Url type="text/html" template="https://elebila.elhuyar.eus/search/">
```
Comment (pocmo) [inline components/browser/search/src/main/assets/searchplugins/elebila.xml:9]: With this one I see a security warning in Firefox desktop.

**Review by pocmo — COMMENTED** [review]
Thank you for this patch.

Can you help us understand what kind of testing you did? :)

File: components/browser/search/src/main/assets/searchplugins/daum-kr.xml (hunk 1 above)
Comment (Debanshu777) [reply components/browser/search/src/main/assets/searchplugins/daum-kr.xml:9]: I was a bit unsure about how to resolve the issue should I remove it or not.
Ps: 1st PR on android-components sorry about that

File: components/browser/search/src/main/assets/searchplugins/elebila.xml (hunk 2 above)
Comment (Debanshu777) [reply components/browser/search/src/main/assets/searchplugins/elebila.xml:9]: In this case, should keep it http?

**Comment by Debanshu777** [comment]
@pocmo Should I remove both the links?
First, one can be resolved neither in HTTPS nor in HTTP. Second is giving an unsafe warning.

**Comment by gabrielluong** [comment]
**🚧 Pull Request Backlog Cleanup**: Closing this since it has been awhile and seems like the changes have gone stale.

**Pull request closed without merging**